internal/bminventory/inventory.go

```diff
@@ -925,8 +923,7 @@ func (b *bareMetalInventory) GenerateClusterISOInternal(ctx context.Context, par
 }
 
 func (b *bareMetalInventory) getIgnitionConfigForLogging(cluster *common.Cluster, params installer.GenerateClusterISOParams, log logrus.FieldLogger) string {
-	ignitionConfigForLogging, _ := b.IgnitionBuilder.FormatDiscoveryIgnitionFile(cluster,
-		b.IgnitionConfig, params.ImageCreateParams, true, b.authHandler.AuthType())
+	ignitionConfigForLogging, _ := b.IgnitionBuilder.FormatDiscoveryIgnitionFile(cluster, b.IgnitionConfig, true, b.authHandler.AuthType())
```

Comment:
the image info, that includes image type is updated later in the code of GenerateClusterISO, so you need to update it earlier  since we want to use the data from the cluster

Reply:
Good catch. I'll update only the cluster object before a call to to this method and let updateImageInfoPostUpload() be responsible for setting it in the DB, with the rest of parameters.

Reply:
I don't think this was fixed.
`updateImageInfoPostUpload` changes more than just the type. This would also be missing the download url, right?

Reply:
It was fixed by adding line [912](https://github.com/openshift/assisted-service/pull/1355/files#diff-747a429a0e12307ef65d754c83b6898ce3858b621005e1e7af6e420da1f1657cR912) 
in order to set the image type on the cluster.image_params. The image type will be persisted to the DB on successful ISO generation by `updateImageInfoPostUpload()`. However for creating the proper discovery ignition, the image-type should be set for the cluster beforehand.

Reply:
Right, but that's just the image type. What about all the other parameters we only set in `updateImageInfoPostUpload`?
For example, the download url.

Reply:
download url isn't needed by `ignition.FormatDiscoveryIgnitionFile`.
It was the imageType which is needed by `ignition.FormatDiscoveryIgnitionFile` to format the ignition.
download url as the rest of the parameters will be updated by `updateImageInfoPostUpload` after the ignition and image are successfully generated, but aren't needed to be set prior to it.

Reply:
Okay, I think I wrote up my concern better in https://github.com/openshift/assisted-service/pull/1355/files#r603518175